api/extension-caql.md

```diff
@@ -0,0 +1,55 @@
+# Issuing CAQL Queries
+
+The Circonus Analytics Query Language (CAQL) allows the user to issue complex queries against metric data residing in IRONdb.
+The [CAQL Reference](https://login.circonus.com/resources/docs/user/caql_reference.html) provides comprehensive documentation about functionality offered by the language.
+
+
+## Description
+
+### URI
+
+`/extension/lua/caql_v1`
+
+### Method
+
+GET
+
+### Inputs
+
+Parameters can be submitted as query string parameters or as JSON payload.
+
+The following parameters are supported:
+
+* `query` : The query to execute.
+
+* `start` : The start time from which to pull data, represented in seconds since the epoch. This value is inclusive (data for the given start time will be pulled).
+
+* `end` : The end time up to which data is pulled, represented in seconds since the epoch. This value is exclusive (data up to, but not including, the given end time will be pulled).
+
+* `period` : The period, in seconds, for which to get data rollups.
+
+* `_timeout` : Spefify a timeout for CAQL processing in seconds. Optional. Default = 4.5.
+
+* `preview` : Boolean parameter ("true"/"false") that enables CAQL preview mode. In preview mode caql processing will be faster but results will only be approximated. Optional. Default = false.
+
+## Examples
+
+```
+echo '{
+  "q":"12+3",
+  "start":1474275000,
+  "end":1474275240,
+  "period":60
+}' | curl http://127.0.0.1:8112/extension/lua/caql_v1 --data @-
+```
+
```

Comment:
Might be helpful if we could see the same query given using query-string parameters.

Reply:
Tiny nit: if someone copy/pastes the query-string example, the URL will need to be single-quoted to protect the shell from `?` and `&`.

Reply:
That is true.